linter: `jsdoc/require-property`

### What version of Oxlint are you using?

0.16.4

### What command did you run?

`oxlint --config ./.oxlintrc.json5`

### What does your `.oxlintrc.json` config file look like?

<details>
<summary><code>.oxlintrc.json5</code></summary>

```json5
{
	"ignorePatterns": [
		"**/fixtures/**",
		"**/fixture.*",
		"**/*.snap.cjs",
	],
	"categories": {
		"correctness": "error",
		"perf": "error",
		"pedantic": "error",
		"suspicious": "warn",
	},
	"plugins": [
		"import",
		"jsdoc",
		"jsx-a11y",
		"node",
		"oxc",
		"react-perf",
		"react",
		"recommended",
		"typescript",
		"unicorn",
	],
	"rules": {
		"eslint/no-case-declarations": "off",
		"eslint/no-constructor-return": "off",
		"eslint/require-await": "off",
		"import/max-dependencies": "off",
		"eslint/max-depth": ["error", 3],
		"eslint/max-lines-per-function": ["error", 30],
		"eslint/max-nested-callbacks": ["error", 3],
		"jsdoc/require-returns": "off",
		"jsdoc/require-returns-description": "off",
		"no-console": "error",
		"no-import-type-side-effects": "error",
		"typescript/ban-types": "off",
		"unicorn/explicit-length-check": "off",
		"unicorn/no-typeof-undefined": "off",
	},
	"overrides": [
		{
			"files": [
				"*.spec.mjs",
				"*.spec.jsx",
				"*.test.mjs",
				"*.test.jsx",
			],
			"rules": {
				"max-depth": "off",
				"max-lines-per-function": "off",
				"max-nested-callbacks": "off",
			},
		},
	],
}
```
</details>

### What happened?

oxlint erroneously errors when the `@prop` shorthand is used

```
  × eslint-plugin-jsdoc(require-property): The `@typedef` and `@namespace` tags must include a `@property` tag with the type Object.
    ╭─[packages/media/media.loader.mjs:46:4]
 45 │  *
 46 │  * @typedef {object} MediaExtensionAddRemoveConfig
    ·    ─────────────────
 47 │  * @prop {FileExtensionsList} MediaExtensionAddRemoveConfig.additions A list of file extensions to add to the default list.
    ╰────
  help: Consider adding a `@property` tag or replacing it with a more specific type.
```

I think this is working as expected, unless we want to diverge on the behavior here from ESLint. I just tried this out of the box with `eslint@9.24.0` and `eslint-plugin-jsdoc@50.6.9` with the following `eslint.config.js`:

```js
import jsdoc from "eslint-plugin-jsdoc";

const config = [
  // configuration included in plugin
  jsdoc.configs["flat/recommended"],
];

export default config;
```

and the output was:

```text
  1:1  warning  Missing JSDoc @property                                                   jsdoc/require-property
  3:1  warning  Invalid JSDoc tag (preference). Replace "prop" JSDoc tag with "property"  jsdoc/check-tag-names

✖ 2 problems (0 errors, 2 warnings)
  0 errors and 2 warnings potentially fixable with the `--fix` option.
```

for this file:
```js
/**
 * @typedef {object} SomeTypedef
 * @prop {string} propName Prop description
 */
```

Although, I don't feel like this is a very productive default to me (if I were designing this plugin), since it seems like it is just emitting errors needlessly. But I'm not sure it's worth diverging on the behavior here currently.

--- a/crates/oxc_linter/src/rules/jsdoc/require_property.rs
+++ b/crates/oxc_linter/src/rules/jsdoc/require_property.rs
@@ -25,6 +25,8 @@ declare_oxc_lint!(
     /// Requires that all `@typedef` and `@namespace` tags have `@property` tags
     /// when their type is a plain `object`, `Object`, or `PlainObject`.
     ///
+    /// Note: this rule can be configured via [jsdoc settings](https://oxc.rs/docs/guide/usage/linter/config-file-reference.html#settings) option.
+    ///
     /// ### Why is this bad?
     ///
     /// Object type should have properties defined.

